### PR TITLE
Allow using array callables for service extension.

### PR DESCRIPTION
**Pull request by curry684**
Ref #201

### CODE REVIEW

**Comment by curry684** [comment]
My PHP low level knowledge isn't to the point where I can reliably fix the C extension for this patch - someone else will have to.

**Comment by sergiors** [comment]
I think these lines can be use `is_callable` also

https://github.com/silexphp/Pimple/pull/202/files#diff-daa3407539df55e8a6514b8286e244acR182
https://github.com/silexphp/Pimple/pull/202/files#diff-daa3407539df55e8a6514b8286e244acR160

**Comment by curry684** [comment]
If I recall correctly I tried those and reverted them because it broke more or less the entire codebase, ref. https://github.com/silexphp/Pimple/issues/201#issuecomment-221112200

Should be part of a different PR imo.

File: src/Pimple/Container.php [hunk 1]
```diff
@@ -233,8 +233,8 @@ public function extend($id, $callable)
             throw new \InvalidArgumentException(sprintf('Identifier "%s" does not contain an object definition.', $id));
         }
 
-        if (!is_object($callable) || !method_exists($callable, '__invoke')) {
-            throw new \InvalidArgumentException('Extension service definition is not a Closure or invokable object.');
+        if (!is_callable($callable)) {
```
Comment (fabpot) [inline src/Pimple/Container.php]: Here is why this is not done this way: we are storing parameters and services i in the same array. Using `is_callable` means that all of a sudden, a simple string store as a parameter can be interpreted as being a service (any built-in PHP function for instance). So, that's not possible. That's the reason why Pimple wants services to be defined as closures.

Comment (curry684) [inline src/Pimple/Container.php]: Valid point, even though 'not possible' is more of a 'could cause too many unintended side effects'.  I can see the rationale in forbidding plain callables, but perhaps that should be documented a bit more explicitly.

**Pull request closed without merging**